src/opentaskpy/addons/gcp/remotehandlers/bucket.py

```diff
@@ -0,0 +1,304 @@
+"""GCP Cloud Bucket remote handler."""
+
+import glob
+import re
+
+import opentaskpy.otflogging
+import requests
+from opentaskpy.exceptions import RemoteTransferError
+from opentaskpy.remotehandlers.remotehandler import RemoteTransferHandler
+
+from .creds import get_access_token
+
+
+class BucketTransfer(RemoteTransferHandler):
+    """GCP CloudBucket remote transfer handler."""
+
+    TASK_TYPE = "T"
+
+    def __init__(self, spec: dict):
+        """Initialise the CloudBucket handler.
+
+        Args:
+            spec (dict): The spec for the transfer. This is either the source, or the
+            destination spec.
+        """
+        self.logger = opentaskpy.otflogging.init_logging(
+            __name__, spec["task_id"], self.TASK_TYPE
+        )
+        super().__init__(spec)
+
+        # Generating Access Token for Transfer
+        self.credentials = get_access_token(self.spec["protocol"])
+
+    def supports_direct_transfer(self) -> bool:
+        """Return False, as all files should go via the worker."""
+        return False
+
+    def handle_post_copy_action(self, files: list[str]) -> int:
+        """Handle the post copy action specified in the config.
+
+        Args:
+            files (list[str]): A list of files that need to be handled.
+
+        Returns:
+            int: 0 if successful, 1 if not.
+        """
+        if "postCopyAction" in self.spec and (
+            self.spec["postCopyAction"]["action"] == "move"
+            or self.spec["postCopyAction"]["action"] == "rename"
+        ):
+            try:
+                # Append a directory if one is defined
+
+                for file in files:
+                    srcFile_encoded = file.replace("/", "%2F")
```

Comment:
snake-case variable names everywhere where, just to maintain variable naming consistency.

Reply:
You snuck one into `creds.py` too

Reply:
I'd name this `source_file_encoded` and `dest_file_encoded`